From c0cb349cb8a27bc384f9e0b68ed95ebace9faa60 Mon Sep 17 00:00:00 2001
From: Jan Blackquill <uhhadd@gmail.com>
Date: Thu, 10 Feb 2022 19:12:53 -0500
Subject: [PATCH] effects/DesktopGrid: use delta, not absolute time for
 MotionManager

Despite the argument naming, the input for WindowMotionManager::calculate is supposed to be a
delta, not an absolute time. Giving it a delta fixes the PresentWindows in the DesktopGrid.

BUG: 443971
---
 src/effects/desktopgrid/desktopgrid.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/effects/desktopgrid/desktopgrid.cpp b/src/effects/desktopgrid/desktopgrid.cpp
index e444d9ff4c..900b49af0f 100644
--- a/src/effects/desktopgrid/desktopgrid.cpp
+++ b/src/effects/desktopgrid/desktopgrid.cpp
@@ -222,7 +222,7 @@ void DesktopGridEffect::prePaintScreen(ScreenPrePaintData& data, std::chrono::mi
         if (isUsingPresentWindows()) {
             for (auto i = m_managers.begin(); i != m_managers.end(); ++i) {
                 for (WindowMotionManager &manager : *i) {
-                    manager.calculate(timeline.currentTime());
+                    manager.calculate(time);
                 }
             }
         }
-- 
GitLab

